### PR TITLE
Implemented auto-discovery of scheme (partition-key, clustering-key, …

### PR DESCRIPTION
**Pull request by pravinbhat**
**What this PR does**:
Implemented auto-discovery of schema (partition-key, clustering-key, column-names) including identification of collections, UDTs & Frozens types. Also find the appropriate TTL & WriteTime columns.

**Which issue(s) this PR fixes**:
Fixes # CDM-3

**Checklist:**
- [ ] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/datastax/astra/migrate/schema/ColumnInfo.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        this.isClusteringKey = isClusteringKey;
+        isNonKey = !isPartitionKey && !isClusteringKey;
+        if (typeInfo.getTypeClass().equals(List.class) || typeInfo.getTypeClass().equals(Set.class)
+                || typeInfo.getTypeClass().equals(Map.class)) {
+            isCollection = true;
+        }
+        if (typeInfo.getTypeClass().equals(UdtValue.class)) {
+            isUDT = true;
+        }
+        if (cm.getType().toString().contains(", frozen")) {
```
Comment (msmygit) [inline src/main/java/datastax/astra/migrate/schema/ColumnInfo.java:35]:
```suggestion
        if (cm.getType().toString().toLowerCase(Locale.ROOT).contains(", frozen")) {
```

File: src/resources/runCommands.txt [hunk 2]
```diff
@@ -4,14 +4,16 @@ curl -OL https://downloads.datastax.com/enterprise/cqlsh-astra.tar.gz
 wget https://archive.apache.org/dist/spark/spark-3.3.1/spark-3.3.1-bin-hadoop3.tgz
 
 // Migrate
-spark-submit --properties-file /<path>/cdm.properties --verbose --master "local[*]" --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.*.jar
-spark-submit --properties-file /<path>/cdm.properties --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.*.jar &> table_out.log
+spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspaceTable="keyspace.table" --master "local[*]" --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.4.*.jar
```
Comment (msmygit) [inline src/resources/runCommands.txt:7]:
```suggestion
spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspace.table="keyspace.table" --master "local[*]" --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.4.*.jar
```

File: src/resources/runCommands.txt [hunk 3]
```diff
@@ -4,14 +4,16 @@ curl -OL https://downloads.datastax.com/enterprise/cqlsh-astra.tar.gz
...
+spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspaceTable="keyspace.table" --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.4.*.jar &> table_out.log
```
Comment (msmygit) [inline src/resources/runCommands.txt:8]:
```suggestion
spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspace.table="keyspace.table" --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.4.*.jar &> table_out.log
```

File: src/resources/runCommands.txt [hunk 4]
```diff
@@ -4,14 +4,16 @@ curl -OL https://downloads.datastax.com/enterprise/cqlsh-astra.tar.gz
...
+// If target keyspace oand/or table name is different than origin, then add --conf spark.target.keyspaceTable="keyspace2.table2"
+// Add option --verbose for verbose output
 
 // Random Partitioner Run Command
-spark-submit --properties-file /<path>/cdm.properties --verbose --master "local[*]" --conf spark.origin.minPartition=-1 --conf spark.origin.maxPartition=170141183460469231731687303715884105728 --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.*.jar
+spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspaceTable="keyspace.table" --master "local[*]" --conf spark.origin.minPartition=-1 --conf spark.origin.maxPartition=170141183460469231731687303715884105728 --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.4.*.jar
```
Comment (msmygit) [inline src/resources/runCommands.txt:13]:
```suggestion
spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspace.table="keyspace.table" --master "local[*]" --conf spark.origin.minPartition=-1 --conf spark.origin.maxPartition=170141183460469231731687303715884105728 --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.4.*.jar
```

Comment (msmygit) [inline src/resources/runCommands.txt:13]:
```suggestion
spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspace.table="keyspace.table" --master "local[*]" --conf spark.origin.minPartition=-1 --conf spark.origin.maxPartition=170141183460469231731687303715884105728 --class datastax.astra.migrate.Migrate /<path>/cassandra-data-migrator-3.4.*.jar
```

File: src/resources/runCommands.txt [hunk 5]
```diff
@@ -4,14 +4,16 @@ curl -OL https://downloads.datastax.com/enterprise/cqlsh-astra.tar.gz
...
 
 // Validate
-spark-submit --properties-file /<path>/cdm.properties --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.DiffData /<path>/cassandra-data-migrator-3.*.jar &> table_out.log
+spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspaceTable="keyspace.table" --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.DiffData /<path>/cassandra-data-migrator-3.4.*.jar &> table_out.log
```
Comment (msmygit) [inline src/resources/runCommands.txt:16]:
```suggestion
spark-submit --properties-file /<path>/cdm.properties --conf spark.origin.keyspace.table="keyspace.table" --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.DiffData /<path>/cassandra-data-migrator-3.4.*.jar &> table_out.log
```

File: src/resources/runCommands.txt [hunk 6]
```diff
@@ -4,14 +4,16 @@ curl -OL https://downloads.datastax.com/enterprise/cqlsh-astra.tar.gz
...
 
 // Guardrail check (identify large fields)
-spark-submit --properties-file /<path>/cdmGuardrail.properties --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.Guardrail /<path>/cassandra-data-migrator-3.*.jar &> table_out.log
+spark-submit --properties-file /<path>/cdmGuardrail.properties --conf spark.origin.keyspaceTable="keyspace.table" --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.Guardrail /<path>/cassandra-data-migrator-3.4.*.jar &> table_out.log
```
Comment (msmygit) [inline src/resources/runCommands.txt:19]:
```suggestion
spark-submit --properties-file /<path>/cdmGuardrail.properties --conf spark.origin.keyspace.table="keyspace.table" --master "local[*]" --driver-memory 25G --executor-memory 25G --class datastax.astra.migrate.Guardrail /<path>/cassandra-data-migrator-3.4.*.jar &> table_out.log
```

**Review by msmygit — COMMENTED** [review]
LGTM. I've left trivial comments with changes.

**Review by msmygit — APPROVED** [review]

**Comment by msmygit** [comment]
Fixes #30, #31, #32 and #33 ?

File: README.md [hunk 7]
```diff
@@ -31,6 +31,7 @@ tar -xvzf spark-3.3.1-bin-hadoop3.tgz
 
 ```
 ./spark-submit --properties-file cdm.properties /
+--conf spark.origin.keyspaceTable="<keyspace-name>.<table-name>" /
```
[the remark this answers is not in the document]
Comment (pravinbhat) [reply README.md:34]: @msmygit I agree the prop name `keyspaceTable` is not great & we should change that, but wont be doing that in this PR due to below reasons
- Breaks backword compatibility & the customers who we might provide this update may not like it
- The suggested change `keyspace.table` may also not be great as we use the dot notation only for namespace & in this case `keyspace` is not a namespace but actually part of property. Instead something like `keyspaceAndTable` OR `keyspaceDotTable` may work better. 
Lets discuss this & implement as a separate PR if thats OK.

File: README.md [hunk 8]
```diff
@@ -116,6 +122,9 @@ This mode is specifically useful to processes a subset of partition-ranges that
 - Validate migration accuracy and performance using a smaller randomized data-set
 - Supports adding custom fixed `writetime`
 
+# Known Limitations
+- This tool does not migrate `ttl` & `writetime` for every field-level (for optimization reasons), but it instead it finds the field with the highest `ttl` & `writetime` and uses those values for `ttl` & `writetime` on the entire Target row
```
[the remark this answers is not in the document]
Comment (pravinbhat) [reply README.md:126]: @msmygit Looks good, will update accordingly.

File: src/main/java/datastax/astra/migrate/schema/ColumnInfo.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    private boolean isUDT;
+    private boolean isFrozen;
+
+    ColumnInfo(ColumnMetadata cm, boolean isPartitionKey, boolean isClusteringKey) {
+        this.colName = cm.getName().toString();
+        typeInfo = new TypeInfo(cm.getType());
+        this.isPartitionKey = isPartitionKey;
+        this.isClusteringKey = isClusteringKey;
+        isNonKey = !isPartitionKey && !isClusteringKey;
+        if (typeInfo.getTypeClass().equals(List.class) || typeInfo.getTypeClass().equals(Set.class)
```
[the remark this answers is not in the document]
Comment (pravinbhat) [reply src/main/java/datastax/astra/migrate/schema/ColumnInfo.java:28]: @msmygit In C* Tuple is not a collection type. From a CDM perspective, we are able to successfully extract & migrate it by just using the Tuple class w/o the need to identify the inner types.

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (msmygit) [reply README.md:34]: I thought we're newly introducing this property as the old one was `spark.origin|target.keyspaceTable`

**Pull request merged**